localstack/services/apigateway/invocations.py

```diff
@@ -215,14 +217,28 @@ def apply_response_parameters(invocation_context: ApiInvocationContext):
     int_responses = integration.get("integrationResponses") or {}
     if not int_responses:
         return response
+
+    integration_type_orig = integration.get("type") or integration.get("integrationType") or ""
+    integration_type = integration_type_orig.upper()
+    if integration_type == "AWS_PROXY":
+        LOG.warning("AWS_PROXY integration type should not apply response parameters")
+        return response
+
     entries = list(int_responses.keys())
     return_code = str(response.status_code)
-    if return_code not in entries:
-        if len(entries) > 1:
-            LOG.info("Found multiple integration response status codes: %s", entries)
-            return response
-        return_code = entries[0]
-    response_params = int_responses[return_code].get("responseParameters", {})
+    # Selecting the right integration response
```

Comment:
What you did is great and will be included in ongoing work to wrap this logic inside `ResponseTemplates`. The class already exists but not applied on every integration.

This is an example of code that should go away because the template handling would have been done inside ResponseTemplates